### DISCUSSION TITLE
Homogenization for non regular and non orthogonal mesh

### DISCUSSION THREAD

**Discussion by ghost** [idaholab/moose#23235] [Q&A Modules: Solid mechanics] [answered]
I am trying to perform asymptotic homogenization of a periodic structure wherein the unit cell is a rhombus. Is there a way to run the homogenization process using the `AsymptoticExpansionHomogenizationKernel`?

**Comment by GiudGiud** [comment]
@jasondhales you're the author of that object

**Comment by jasondhales** [comment, marked as answer]
I haven't tried it, but my guess is that as long as you set up the periodic boundary conditions properly, it will work.

**Reply by ghost** [reply]
Is it possible to change the repeating axes orientation? For example, when defining the periodic BC for a square unit cell, the repeating axes (X and Y) are perpendicular to each other and coincident with the Cartesian plane. However, for a rhombus type unit cell, this wouldn't be case.
